core/src/test/java/hudson/UtilTest.java

```diff
@@ -248,7 +258,22 @@ public void testIsSymlink() throws IOException, InterruptedException {
         }
     }
 
-    @Test public void deleteFile() throws Exception {
+    @Test
+    public void testDeleteFile() throws Exception {
+        File d = tmp.getRoot();
+        File f = new File(d, "f");
+        try {
+            // Test: File is deleted
+            mkfiles(f);
+            Util.deleteFile(f);
+            assertFalse("f exists after calling Util.deleteFile", f.exists());
+        } finally {
+            Util.deleteRecursive(d);
```

Comment:
What is the point in introducing `d`? If the file is created in `tmp` itself there is no need for any teardown.